src/simulator/services/resources/directory.py

```diff
@@ -78,10 +85,21 @@ def _unpickle(file_name: str, directory: str) -> Any:
         file_name = Directory._add_extension(file_name, "pickle")
         if not os.path.isfile(directory + file_name):
             return None
-        handle: BinaryIO
 
-        with open(directory + file_name, 'rb') as handle:
-            return smart_load(handle)
+        return smart_load(directory + file_name)
+
+    @staticmethod
+    def _unjson(file_name: str, directory: str) -> Any:
```

Comment:
perhaps we should add some more clear rules on what is pickled and what is json'd....we currently use pickle for map gen for ML